File: ingest/ingest/models/sensor_module.py
from sqlalchemy import Column, Integer, String, Float, DateTime
from ingest.models.base import Base

class SensorModule(Base):
    __tablename__ = 'sensor_module'
    
    id = Column(Integer, primary_key=True, autoincrement=False)
    name = Column(String, nullable=False)
    latitude = Column(Float, nullable=False)
    longitude = Column(Float, nullable=False)
    last_ping = Column(DateTime, nullable=False)
    created_at = Column(DateTime, nullable=False)

File: web/src/models/sensor-module.ts
export interface SensorModule {
  id: number;
  name: string;
  latitude: number;
  longitude: number;
  last_ping: Date;
  created_at: Date;
}

export const initSensorModule = (): SensorModule => ({
  id: 0,
  name: "",
  latitude: 0,
  longitude: 0,
  last_ping: new Date(),
  created_at: new Date(),
});
